libraries/Kimai/Remote/Database/Civicrm.php

```diff
@@ -299,9 +305,11 @@ public function addQueueDeletedTimesheet($queueCutoff)
      * @param $queueCutoff
      * @return array of queued timesheet
      */
-    public function getQueuedData($limit)
+    public function getQueuedData()
     {
-        $query = "SELECT * FROM `{$this->getCivicrmQueue()}` WHERE `confirmed` IS NULL ORDER BY modified DESC LIMIT {$limit}";
+        $limit = $this->sessionParams['limit'] ?? 25;
+        $projectId = $this->sessionParams['projectId'];
```

Comment:
@lucky091588 For $limit and $projectId, please cast both of these with `(int)`; otherwise we risk sql injection if these are malicious strings.

Also, please default $projectId to 0; otherwise we risk a fatal sql error if it is empty.